tiledb/sm/query/query_condition.h

```diff
@@ -350,18 +356,18 @@ class QueryCondition {
    * @return The filtered cell slabs.
    */
   template <typename T, QueryConditionOp Op>
-  std::vector<ResultCellSlab> apply_clause(
-      const Clause& clause,
+  std::vector<ResultCellSlab> apply_val(
+      const ASTNodeVal* node,
```

Comment:
Why is this not `apply_ast` taking an ASTNode?  Also, why is it taking const pointer rather than a const reference?  Same question for all `apply_val_*` below.

Reply:
I think if it took a const reference it could be fine; it takes a pointer currently because I'm more used to programming in C LOL

Reply:
Resolved locally

Reply:
Have to break any C habits.  Just forget that C++ ever had anything to do with C.